src/features/query-bar/components/filter/FilterButton.js

```diff
@@ -30,16 +32,25 @@ type Props = {
 class FilterButton extends React.Component<Props, State> {
     constructor(props: Props) {
         super(props);
-        const conditionID = this.generateConditionID();
-        const initialCondition = this.createCondition(props, conditionID);
+
         this.state = {
             appliedConditions: [],
-            conditions: [initialCondition],
+            conditions: [],
+            selectedConnector: AND,
             areErrorsEnabled: false,
             isMenuOpen: false,
         };
     }
 
+    componentDidMount() {
+        const id = this.generateConditionID();
```

Comment:
generateConditionID and createCondition seem coupled. Any reasons not to merge into one?

Reply:
No it makes more sense to just reference id within createCondition